ogr/ogrsf_frmts/mssqlspatial/ogrmssqlspatialdatasource.cpp

```diff
@@ -714,6 +717,18 @@ int OGRMSSQLSpatialDataSource::Open( const char * pszNewName, bool bUpdate,
             nCurrent, nNext, nTerm, FALSE))
             continue;
 
+        if (ParseValue(&pszUID, pszConnectionName, "uid=",
+            nCurrent, nNext, nTerm, FALSE))
+            continue;
+
+        if (ParseValue(&pszPWD, pszConnectionName, "pwd=",
+            nCurrent, nNext, nTerm, FALSE))
+            continue;
+
+        if (ParseValue(&pszTrustedConnection, pszConnectionName, "trustedconnection=",
```

Comment:
Originally, I planned not to query `UID` and `PWD` if `TrustedConnection` is specified in the connection string. But then I thought, that if env variables are set, that has been deliberate too. So, if UID and PWD are defined in the environment variables, it may be safe to add it anyway. 
Alternatively, the block where UID and PWD are injected in the connection string can be only executed only if the connection string dies not contain `TrustedConnection`... Let me know what you prefer...

Reply:
> Let me know what you prefer...

no idea. I'm not familiar at all with MSSQLSpatial authentication. But I don't really understand your comment: I don't see anything in the code (latest version) that tests for "trustedconnection".

Reply:
Sorry for not being clear. yes, with the latest commit I removed the check **trusted_connection**. In the comment I just wanted to clarify, why I added it at first...

I am no MS SQL expert either. A second look at the authentication documentation tells me, that **trusted_connection**  can also be stored in the _odbc.ini_ as part of the _DSN_. So it will not be part of the connection string.

Therefore, I think the current implementation that does not check for that keyword is appropriate. If a user creates the **MSSQLSPATIAL_UID** and **MSSQLSPATIAL_PWD** environment variables, they are used and get priority over **trusted_connection**.

A bit more clarification in the docs can be useful. Will have a look at that...